### PR TITLE
Egress router & local network

### PR DESCRIPTION
**Pull request by fgiloux**
Adding a test for not having a route added if the gateway and the destination addresses are the same. Confer issue [#12248](https://github.com/openshift/origin/issues/12248)

### CODE REVIEW

File: images/router/egress/egress-router.sh [hunk 1]
```diff
@@ -21,8 +21,9 @@ if ! ip route get ${EGRESS_DESTINATION} | grep -q macvlan0; then
     ip addr add ${EGRESS_SOURCE}/32 dev macvlan0
     ip link set up dev macvlan0
     ip route add ${EGRESS_GATEWAY}/32 dev macvlan0
-    ip route add ${EGRESS_DESTINATION}/32 via ${EGRESS_GATEWAY} dev macvlan0
-
+    if [ ! "${EGRESS_DESTINATION}" == "${EGRESS_GATEWAY}" ]; then
```
Comment (stevekuznetsov) [inline images/router/egress/egress-router.sh]: Does the `!=` syntax work with `[`? If so, it may read better:

```sh
if [ "${EGRESS_DESTINATION}" != "${EGRESS_GATEWAY}" ]; then
```

File: images/router/egress/egress-router.sh [hunk 2]
```diff
@@ -21,8 +21,9 @@ if ! ip route get ${EGRESS_DESTINATION} | grep -q macvlan0; then
...
+        ip route add ${EGRESS_DESTINATION}/32 via ${EGRESS_GATEWAY} dev macvlan0
```
Comment (stevekuznetsov) [inline images/router/egress/egress-router.sh]:
```sh
ip route add "${EGRESS_DESTINATION}/32" via "${EGRESS_GATEWAY}" dev macvlan0
             ^                        ^     ^                 ^
```

**Comment by fgiloux** [comment]
Yes fine. I am also adding parameter escaping to the original code. It is not a big thing here as there is no risk of privilege escalation: the user giving the parameters needs the rights for creating the pod.

**Comment by knobunc** [comment]
[test]

**Comment by knobunc** [comment]
@danwinship please can you check to see if this makes sense.

**Comment by danwinship** [comment]
So... you're using the egress router to talk to a server on the same subnet as your OpenShift cluster?

**Comment by fgiloux** [comment]
Yes, it may be a bit of an edge case, mostly in test environments where all servers are in a single subnet and IP address white listing is used for instance with MySQL.

**Comment by danwinship** [comment]
OK, I guess that makes sense. Please squash the two commits together though, and rebase it to master rather than having a merge commit

**Comment by danwinship** [comment]
[merge]
